src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -149,7 +152,7 @@ private static string GetDirectoryName(string path, bool isUnixLike)
             return null;
         }
 
-        internal static bool IsSameDirectoryOrChildOf(string child, string parent)
+        internal static bool IsSameDirectoryOrChildOf([DisallowNull]string? child, string parent)
```

Comment:
This would be much clearer if you made a temp `string?` variable. This exposes private implementation details of `IsSameDirectoryOrChildOf` to callers, which seems like bad practice.